src/OrleansRuntime/GrainDirectory/LocalGrainDirectory.cs

```diff
@@ -471,12 +479,17 @@ public SiloAddress CalculateTargetSilo(GrainId grainId, bool excludeThisSiloIfSt
                     return excludeThisSiloIfStopping && !Running ? null : MyAddress;
                 }
 
-                // excludeMySelf from being a TargetSilo if we're not running and the excludeThisSIloIfStopping flag is true. see the comment in the Stop method.
-                bool excludeMySelf = !Running && excludeThisSiloIfStopping; 
-
                 // need to implement a binary search, but for now simply traverse the list of silos sorted by their hashes
```

Comment:
The binary search will provide benefit only on relatively large deployments (with silos count > 75). Does it makes sense to implement this?